CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(libNEGF VERSION 0.1 LANGUAGES Fortran C)
+
+set(FYPP "fypp" CACHE STRING "Fypp preprocessor")
+
+option(WITH_MPI "Whether MPI-parallelised library should be built" FALSE)
+
+# Note, this option does not work yet, so leave it to false
+option(WITH_INELASTIC "Whether to build with inelastic scattering" FALSE)
+
+set(MPIFX_INCLUDE_DIR "/opt/mpifx/include" CACHE PATH
+  "Include directory of the MPIFX library (only if built with MPI)")
+
+set(INSTALL_LIB_DIR "${CMAKE_INSTALL_PREFIX}/lib" CACHE PATH
+  "Installation directory for libraries")
+
+set(INSTALL_MOD_DIR "${CMAKE_INSTALL_PREFIX}/include/libnegf/modfiles" CACHE PATH
+  "Installation directory for Fortran module files")
+
+option(BUILD_SHARED_LIBS "Whether the library should be shared" FALSE)
```

Comment:
I got a bit in troubles with ext_sparsekit and ext_system setting this. I solved by forcing static compilation of those two in their cmake files, and adding here:
```
if(BUILD_SHARED_LIBS)
  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
endif()
```

Reply:
Ok, I just wasn't linking my test application to libsparsekit and libsyscalls. However, even if they are labeled as external libs, I think it makes more sense to have ext_sparsekit included  in libnegf.so (no one else uses it). And maybe also ext_system, or you use it separately in dftb?
In the above, I tried to do it in a weird way, linking static libraries to the dynamic one. I think the `OBJECT` attribute in `add_library` would do it in the right way. We can skip this for this PR.